### PR TITLE
Do internal restart if provisioner is set

### PR DESCRIPTION
**Pull request by freva**

### CODE REVIEW

File: config-provisioning/src/main/java/com/yahoo/config/provision/HostFilter.java [hunk 1]
```diff
@@ -27,15 +27,11 @@ private HostFilter(Set<String> hostnames,
...
-
-        this.hostnames = hostnames;
-        this.flavors = flavors;
-        this.clusterTypes = clusterTypes;
-        this.clusterIds = clusterIds;
+        this.hostnames = Objects.requireNonNull(hostnames, "Hostnames cannot be null, use an empty list");
+        this.flavors = Objects.requireNonNull(flavors, "Flavors cannot be null, use an empty list");
+        this.clusterTypes = Objects.requireNonNull(clusterTypes, "clusterTypes cannot be null, use an empty list");
+        this.clusterIds = Objects.requireNonNull(clusterIds, "clusterIds cannot be null, use an empty list");
+        System.out.println(hostnames);
```
Comment (hmusum) [inline config-provisioning/src/main/java/com/yahoo/config/provision/HostFilter.java:34]: Remove debug line?

**Review by hmusum — APPROVED** [review]
One comment, otherwise LGTM

File: config-provisioning/src/main/java/com/yahoo/config/provision/HostFilter.java (hunk 1 above)
Comment (freva) [inline config-provisioning/src/main/java/com/yahoo/config/provision/HostFilter.java:34]:
```suggestion
```

Comment (freva) [reply config-provisioning/src/main/java/com/yahoo/config/provision/HostFilter.java]: Thanks :+1:

**Pull request merged**